examples/vpc_service_control/vpc_sc_project/main.tf

```diff
@@ -0,0 +1,318 @@
+/*********************************/
+
+resource "google_compute_firewall" "allow_all_from_internal" {
+  allow {
+    protocol = "all"
+  }
+
+  direction     = "INGRESS"
+  disabled      = false
+  name          = "allow-all-from-internal"
+  network       = "${google_compute_network.test-vpc.self_link}"
+  priority      = "1000"
+  project       = "${google_project.vpc_sc_network_project.project_id}"
+  source_ranges = ["10.0.0.0/8"]
+}
+
+
+
+
+
+/*************************/
+/***** BEGIN ROUTES ******/
+/*************************/
+
+resource "google_compute_route" "google_private_access_route" {
+  dest_range       = "199.36.153.4/30"
+  name             = "google-private-access-route"
+  network          = "${google_compute_network.test-vpc.self_link}"
+  next_hop_gateway = "https://www.googleapis.com/compute/v1/projects/${google_project.vpc_sc_network_project.project_id}/global/gateways/default-internet-gateway"
+  priority         = "1000"
+  project          = "${google_project.vpc_sc_network_project.project_id}"
+}
+
+
+
+
+/******************************/
+/***** BEGIN PRIVATE DNS ******/
+/******************************/
+/* STEP 7A: DELETE THIS COMMENT LINE TO CREATE DNS RECORDS THAT RESOLVE *.GOOGLEAPIS.COM TO GOOGLE PRIVATE ACCESS 
+// Steps below are reproduced from gcloud comands here:
+// https://cloud.google.com/vpc-service-controls/docs/set-up-private-connectivity#configuring_dns_with 
+
+
+resource "google_dns_managed_zone" "google_private_access_zone" {
+  name = "google-private-access-zone"
+  dns_name = "googleapis.com."
+  project          = "${google_project.vpc_sc_network_project.project_id}"
+  description = "Private DNS zone for resolving queries to Google APIs using Google Private Access"
+
+  visibility = "private"
+
+  private_visibility_config {
+    networks {
+      network_url =  "${google_compute_network.test-vpc.self_link}"
+    }
+  }
+}
+
+resource "google_dns_record_set" "cname" {
+  name = "*.googleapis.com."
+  managed_zone = "${google_dns_managed_zone.google_private_access_zone.name}"
+  type = "CNAME"
+  ttl  = 300
+  rrdatas = ["restricted.googleapis.com."]
+  project          = "${google_project.vpc_sc_network_project.project_id}"
+}
+
+
+resource "google_dns_record_set" "a" {
+  name = "restricted.googleapis.com."
+  managed_zone = "${google_dns_managed_zone.google_private_access_zone.name}"
+  type = "A"
+  ttl  = 300
+  rrdatas = ["199.36.153.4", "199.36.153.5", "199.36.153.6", "199.36.153.7"]
+  project          = "${google_project.vpc_sc_network_project.project_id}"
+}
+ STEP 7B: DELETE THIS END-COMMENT LINE.  Then run terraform apply. */
+
+
+
+
+/***************************************/
+/***** BEGIN VPC SERVICE CONTROLS ******/
+/***************************************/
+/* STEP 8A: DELETE THIS COMMENT LINE TO ENABLE VPC SERVICE CONTROLS.  OR - USE THE SECURITY SECTION OF THE CLOUD CONSOLE TO ENABLE THEM  
+
+resource "google_access_context_manager_service_perimeter" "service_perimeter" {
```

Comment:
This should be using the [regular_service_perimeter](https://github.com/terraform-google-modules/terraform-google-vpc-service-controls/tree/master/modules/regular_service_perimeter) submodule to be an example of the module.

Reply:
Perhaps we can do this in release 2.

Reply:
Unfortunately no, this a hard requirement for accepting this Pull request.
This repo is a VPC SC Terraform module. Examples should be at least show how to use the module.

Reply:
Done.